### PR TITLE
Organisations

### LINKED ISSUE #428: [quest] we need "organisations
So we need organisations.

- [ ] add an orgs model

which should have 
 - org name (like Nagarro)
 - org full name ( like Nagarro Software Pvt. Ltd.)
 - org email domain (something like nagarro.com) (optional)
 - org website 

----

- [ ] many-to-many mapping between orgs and users as OrgAdmins 
so that we can have multiple admins for every org

----

- [ ] another many-to-many mapping between orgs and users called OrgMembers
the org-members table should have three columns

- orgId (ofcourse)
- userId (ofcourse)
- email

### PR DESCRIPTION
**Pull request by VibhorCodecianGupta**
fixes #428 

@championswimmer @witty123 @abhishek97 done

### CODE REVIEW

**Comment by championswimmer** [comment]
This pull request **introduces 2 alerts** when merging 1ae75b8b5c90cefc32ffd8c4c76843560f016c40 into ec87daf6718d4de54b4194dbfb4503fd643fa25c - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-5922c9a1a57c2687b51244c585a933ef077f576b)

**new alerts:**

* 2 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by championswimmer** [comment]
This pull request **introduces 2 alerts** when merging 6f91709dc0def21ecd6d4b880cbb7545e7276ece into ec87daf6718d4de54b4194dbfb4503fd643fa25c - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-9c6741f7764187edf0e5e1886f1bf9eb76417d2d)

**new alerts:**

* 2 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by abhishek97** [comment]
@VibhorCodecianGupta while you're at it, can you please add an optional qp here https://github.com/coding-blocks/oneauth/blob/master/src/routers/api/users.js#L52 to support fetching user's org data.

**Comment by championswimmer** [comment]
This pull request **introduces 6 alerts** when merging bac90a0e469b2f24391afd305d903c165d1bb5de into ec87daf6718d4de54b4194dbfb4503fd643fa25c - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-ee2088e0ca52402212c23b6811ea51985580a68a)

**new alerts:**

* 6 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by championswimmer** [comment]
This pull request **introduces 5 alerts** when merging e8b7cb5bc426985d3397f5689725f11f06392fed into ec87daf6718d4de54b4194dbfb4503fd643fa25c - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-d45f353042dff9df2748efceca19fa9e3183a862)

**new alerts:**

* 5 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by championswimmer** [comment]
This pull request **introduces 5 alerts** when merging 4421f45e38ab8df9ae3408e8d950a1a338beede9 into aea54d0e80b0ec342cfa5e0516ff6f68aefef720 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-0e76eeb12d91cae0c112d9bf247695a3f03e6b48)

**new alerts:**

* 5 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/controllers/organisation.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    return members
+}
+
+module.exports = {
+  findOrganisationById,
+  findAllOrganisationsByUserId,
+  findAllOrganisations,
+  createOrganisation,
+  updateOrganisation,
+  addOrgAdmin,
```
Comment (championswimmer) [inline src/controllers/organisation.js]: addAdminToOrg would be ideal name for this

File: src/controllers/organisation.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  addOrgMember,
```
Comment (championswimmer) [inline src/controllers/organisation.js]: similarly `addMemberToOrg`

File: src/db/models.js [hunk 3]
```diff
@@ -103,6 +103,53 @@ const Client = db.define('client', {
 Client.belongsTo(User)
 User.hasMany(Client)
 
+const Organisation = db.define('organisation', {
+    id: {type: Sequelize.DataTypes.BIGINT, primaryKey: true},
```
Comment (championswimmer) [inline src/db/models.js]: Only `INTEGER` is fine for id. We won't have 2^32 orgs :D

File: src/db/models.js [hunk 4]
```diff
@@ -103,6 +103,53 @@ const Client = db.define('client', {
...
+    name: {type: Sequelize.DataTypes.STRING, allowNull: false},
+    full_name: {type: Sequelize.DataTypes.STRING, allowNull: false},
+    domain: Sequelize.DataTypes.ARRAY(Sequelize.DataTypes.STRING),
+    website: Sequelize.DataTypes.STRING
+})
+
+const OrgAdmin = db.define('orgadmin', {
+    organisationId: Sequelize.DataTypes.BIGINT,
```
Comment (championswimmer) [inline src/db/models.js]: turn all BIGINT into INTEGER only

**Comment by championswimmer** [comment]
looking good, just do the asked changes, will merge

**Comment by VibhorCodecianGupta** [comment]
@championswimmer made the changes

**Comment by championswimmer** [comment]
This pull request **introduces 2 alerts** when merging 6df059cd866818064168a88af67bfb7fa7b9e17c into 008b037b301275b8440eda750a899b7eab636f9a - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-a8a9d1192621336190b03630b2f4ea64dec1fc4e)

**new alerts:**

* 2 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**